embedded-graphics/src/primitives/line/mod.rs

```diff
@@ -62,6 +70,47 @@ impl Dimensions for Line {
     }
 }
 
+/// Check signs of two signed numbers
+///
+/// Fastest ASM output compared to other methods. See: https://godbolt.org/z/zVx9cD
+fn same_signs(a: i32, b: i32) -> bool {
+    a ^ b >= 0
```

Comment:
While longer, maybe (a >= 0) == (b >= 0) is easier to understand?

https://godbolt.org/z/hsEqPv

Reply:
Good shout. It never ceases to amaze me what compilers can optimise down to.